packages/eyes/src/eyes-test/eyes.ts

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env node
+
+const {TestFailedError, TestResults} = require('@applitools/eyes.sdk.core');
+const {makeVisualGridClient, initConfig} = require('@applitools/visual-grid-client');
+const domNodesToCdt = require('@applitools/visual-grid-client/src/browser-util/domNodesToCdt');
+import * as path from 'path';
+import * as fs from 'fs';
+import glob from 'glob';
+import chalk from 'chalk';
+import {JSDOM} from 'jsdom';
+import {consoleLog} from '@ui-autotools/utils';
+
+const fileNameRegex = /([a-zA-Z]+)(?:@sim)([\d]+)(?:@)([a-zA-Z0-9]+)(?:@variant)([\d]+)(?:@)([a-zA-Z0-9]+)/;
+
+interface IResult {
+  name: string;
+  isNew: boolean;
+  isModified: boolean;
+  url: string;
+  isError?: boolean;
+  error?: any;
+}
+
+function getGridClientConfig(projectPath: string) {
+  const projectName = require(path.join(projectPath, 'package.json')).name;
+  if (!projectName) {
+    throw new Error('The project should have a package.json file with a "name" field.');
+  }
+
+  if (!process.env.APPLITOOLS_API_KEY) {
+    throw new Error('The environment variable "APPLITOOLS_API_KEY" needs to be defined.');
+  }
+
+  const branchName = projectName + '/master';
+  const viewportWidth = 800;
+  const viewportHeight = 600;
+
+  return {
+    appName: projectName,
+    branchName,
+    batchName: projectName,
+    browser: {
+      name: 'chrome',
+      width: viewportWidth,
+      height: viewportHeight
+    }
+  };
+}
+
+interface IResource {
+  url: string;
+  type: string;
+  value: Buffer;
+}
+
+function getStaticResources(cssFilenames: string[], resourceDir: string) {
+  const resources: {[url: string]: IResource} = {};
+  for (const cssFilename of cssFilenames) {
+    const url: string = '/' + cssFilename;
+    resources[url] = {
+      url,
+      type: 'text/css',
+      value: fs.readFileSync(path.join(resourceDir, cssFilename))
+    };
+  }
+  return resources;
+}
+
+function logEyesResult({name, isNew, isModified, url, isError, error}: IResult) {
+  const fileSections = name.match(fileNameRegex);
+  const componentName = fileSections![1];
```

Comment:
Create functions for generating / parsing the filename, and define them next to each other so it's easier to keep them in sync.